src/java/org/apache/cassandra/db/ClusteringComparator.java

```diff
@@ -391,7 +391,24 @@ public String toString()
     public <V> Clustering<?> clusteringFromByteComparable(ValueAccessor<V> accessor,
                                                           ByteComparable comparable)
     {
-        ByteComparable.Version version = ByteComparable.Version.OSS50;
+        return clusteringFromByteComparable(accessor, comparable, ByteComparable.Version.OSS50);
```

Comment:
I wonder what is the reason that a specific version needs to be used?

Callers of this method in `TrieBackedParition` and `TriePartitionUpdater` pass in a `ByteComparable` argument that was `preencoded` using `TrieBackedPartition.BYTE_COMPARABLE_VERSION` (not to be confused with `TypeUtil.BYTE_COMPARABLE_VERSION`).

Maybe all callers should use the new method signature below, always passing the intended `Version`?

Methods `boundFromByteComparable` and `boundaryFromByteComparable` also specifically use `OSS50` - maybe we should consider adding a version argument for those as well? Afaict, these methods are only used by tests?

Reply:
I am not sure which is fully correct there. It looks connected to creating a Peekable instance, but I'm not sure why. Perhaps @blambov can provide more detail.

Reply:
That is, the long term fix might be a bit more nuanced, but I think we'll merge this for now since I know it is correct in the context of getting a clustering column from the ByteComparable object.

Reply:
Dan is right, this is an omission on my part. We should always specify the version for these methods, and the version to use for the trie memtable calls is indeed `TrieBackedPartition.BYTE_COMPARABLE_VERSION`.